lib/msf/core/exploit/remote/http_client.rb

```diff
@@ -373,22 +373,34 @@ def send_request_raw(opts = {}, timeout = 20, disconnect = false)
   # Connects to the server, creates a request, sends the request,
   # reads the response
   #
+  # If a +Msf::Exploit::Remote::HTTP::HttpCookieJar+ instance is passed in the +opts+ dict under a 'cookie' key, said CookieJar will be used in
+  # the request instead of the module +cookie_jar+
+  #
   # Passes `opts` through directly to {Rex::Proto::Http::Client#request_cgi}.
   # Set `opts['keep_cookies']` to keep cookies from responses for reuse in requests.
+  # Cookies returned by the server will be stored in +cookie_jar+
+  #
+  # +expire_cookies+ will control if +cleanup+ is called on any passed +Msf::Exploit::Remote::HTTP::HttpCookieJar+ or the client cookiejar
   #
   # @return (see Rex::Proto::Http::Client#send_recv))
-  def send_request_cgi(opts = {}, timeout = 20, disconnect = true)
-    if cookie_jar.any?
-      opts = { 'cookie' => cookie_jar.to_a.join(' ') }.merge(opts)
+  def send_request_cgi(opts = {}, timeout = 20, disconnect = true, expire_cookies = true)
```

Comment:
This extra argument at the end seems a bit smelly, are there any potential alternative approaches? 🤔 

For instance, if I wanted to set the value to false I'd have to do something like this to disable it:
```
send_request_cgi(..., 20, true, false)
```

Reply:
Agreed, I'll move the `expire_cookies` into `opts`